### PR TITLE
19 edit recipe

### LINKED ISSUE #19: Edit a recipe
Work in source/backend/routes/recipes.js

Write a function that allows a user to edit a recipe given that they are logged in and the creator.
For now, do not worry about if the user is logged in. Make sure that they are the creator and then we can add the check to make sure they are logged in. We will have to add some kind of middleware to check if they are logged in or something along those lines.

PUT /recipes/:recipeId
First check if user is logged in
Check if user has ownership to the recipeId (in recipes table)
Yes:
Change the recipe to the PUT payload
No:
Return 401 (unauthorized)

### PR DESCRIPTION
**Pull request by Solowyhn**
# Resolves #19 

### What were your changes
- I add a put function to source/backend/routes/recipes.js
- Also an updateRecipe function to source/backend/database/model/recipesModel.js

### What specifically should be checked
Check if the update recipes implementation is correct

### CODE REVIEW

File: source/backend/routes/recipes.js [hunk 1]
```diff
@@ -28,6 +29,30 @@ router.get('/recipeId/:recipeId', async (req, res) => {
...
+    const { userId } = req.userInfo;
+    const { recipeId } = req.params;
+
+    const recipe = await recipesModel.getByUserIdAndRecipeId(userId, recipeId);
+    if (recipe.length == 0) {
+      return res.status(401).json({msg: 'Unauthorized to edit recipe'});
+    }
+
+    const updatedRecipe = req.body;
+    await recipesModel.updateByRecipeId(updatedRecipe, recipeId);
```
Comment (joecai6) [inline source/backend/routes/recipes.js:44]: I couldnt find the updateByRecipeId function. did you mean updateRecipe()?

Comment (joecai6) [reply source/backend/routes/recipes.js:44]: You can change the updateRecipe in model to updateByRecipeId

Comment (Solowyhn) [reply source/backend/routes/recipes.js:44]: Yes, I forgot to change the function name accordingly. I will call updateRecipe instead. Thanks for pointing that out.

File: source/backend/database/models/recipesModel.js [hunk 2]
```diff
@@ -46,6 +46,17 @@ async function createRecipe(payload, ingredientsArray) {
...
+ * update recipe by recipe id
+ * @param {*} payload 
+ * @param {*} recipeId 
+ */
+async function updateRecipe(payload, recipeId) {
+  await db('recipes')
+    .update(payload)
+    .where({recipeId});
+}
+
```
Comment (demyinn00) [inline source/backend/database/models/recipesModel.js:59]: This looks good, but I think we need to do a transaction to update the ingredients table as well and potentially the recipeIngredients table (if there are extra/new ingredients)

File: source/backend/routes/recipes.js [hunk 3]
```diff
@@ -28,6 +28,30 @@ router.get('/recipeId/:recipeId', async (req, res) => {
...
+    return res.status(200).json({updatedRecipe, msg: 'Successfully edited a recipe'});
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({
+      message: 'Failed to edit recipe',
+      err,
+    });
+  }
+});
+
```
Comment (demyinn00) [inline source/backend/routes/recipes.js:54]: We will need to modify this according to the above statement.

**Review by demyinn00 — APPROVED** [review]

**Pull request merged**